src/cryptography/hazmat/primitives/kdf/kbkdf.py

```diff
@@ -114,17 +135,27 @@ def derive(self, key_material: bytes, prf_output_size: int) -> bytes:
         if rounds > pow(2, len(r_bin) * 8) - 1:
             raise ValueError("There are too many iterations.")
 
+        fixed = self._generate_fixed_input()
```

Comment:
This *should* be fine to move outside the loop, the fixed input doesn't change across iterations, except for the counter value.